activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1511,6 +1511,49 @@ def test_preloading_has_many_through_with_custom_scope
     Author.preload(:readonly_comments).first!
   end
 
+  # Associations::Preloader#preloaders_on works with hash-like objects
+  test "preloading works when object that responds to :to_hash is passed" do
```

Comment:
Could avoid the passive voice here and write `"preloading works with object that responds to :to_hash"`